### PR TITLE
[Manta-PC] Update dependencies to `v0.9.12`

### LINKED ISSUE #235: Upgrade dependencies to v0.9.12-1
There's a new polkadot release v0.9.12-1
[5feed981cf0fe671447eabaa9c0d235a8dd0003e](https://github.com/paritytech/polkadot/releases/tag/v0.9.12-1)

And related deps:
substrate -> [d76f39995315ec36980908e4b99709bd14927044](https://github.com/paritytech/substrate/tree/polkadot-v0.9.12)
cumulus -> [935bac869a72baef17e46d2ae1abc8c0c650cef5](https://github.com/paritytech/cumulus/tree/polkadot-v0.9.12)

### LINKED ISSUE #239: Clean up compilation warnings

### PR DESCRIPTION
**Pull request by ghzlatarev**
## Description

closes #235
closes #239

### Client

1. version from `3.0.8` to `3.0.9`.

### Runtime

1. set `RelayNetwork` const to correct `Kusama` instead of `Polkadot` in the calamari runtime.
2. `pub type LocalOriginToLocation = ();` means that no one can become `XcmSendOrigin` or `XcmExecuteOrigin` in Calamari/Manta runtimes.
2. `type XcmExecuteFilter = Nothing;` means that even if somehow someone manages to becomes `XcmExecuteOrigin` their message will be filtered out because `Nothing`s implementation of `Contains` trait will always return false.
3. Removed associated type in `pallet_session` because of API change ---> `DisabledValidatorsThreshold` The relevant substrate PR is paritytech/substrate#9448 

## Upstream Changes
Polkadot Changes:
1. https://github.com/paritytech/polkadot/pull/4058
2. https://github.com/paritytech/polkadot/pull/4036
3. https://github.com/paritytech/polkadot/pull/4062
4. https://github.com/paritytech/polkadot/pull/3988
5. https://github.com/paritytech/polkadot/pull/3980
6. https://github.com/paritytech/polkadot/pull/4077
7. https://github.com/paritytech/polkadot/pull/4108
8. https://github.com/paritytech/polkadot/pull/4157
9. https://github.com/paritytech/polkadot/pull/3991
10. https://github.com/paritytech/polkadot/pull/3993
11. https://github.com/paritytech/polkadot/pull/3904
12. https://github.com/paritytech/polkadot/pull/3997
13. https://github.com/paritytech/polkadot/pull/3631
14. https://github.com/paritytech/polkadot/pull/4001
15. https://github.com/paritytech/polkadot/pull/3966
16. https://github.com/paritytech/polkadot/pull/3924
17. https://github.com/paritytech/polkadot/pull/3564
18. https://github.com/paritytech/polkadot/pull/3627
19. https://github.com/paritytech/substrate/pull/9890
20. https://github.com/paritytech/substrate/pull/9788
21. https://github.com/paritytech/polkadot/pull/3815
22. https://github.com/paritytech/polkadot/pull/3911
Release Checklist:
1. https://github.com/paritytech/polkadot/issues/4095

Substrate Changes:
1. https://github.com/paritytech/substrate/pull/9448
2. https://github.com/paritytech/substrate/pull/9971
2. https://github.com/paritytech/substrate/pull/9834
3. https://github.com/paritytech/substrate/pull/9663
4. https://github.com/paritytech/substrate/pull/10013
5. https://github.com/paritytech/substrate/pull/9966
6. https://github.com/paritytech/substrate/pull/9938
7. https://github.com/paritytech/substrate/pull/9789
8. https://github.com/paritytech/substrate/pull/10004
9. https://github.com/paritytech/substrate/pull/9948
10. https://github.com/paritytech/substrate/pull/9755

Cumulus Changes:
1. https://github.com/paritytech/cumulus/pull/651
2. https://github.com/paritytech/cumulus/pull/517
3. https://github.com/paritytech/cumulus/pull/662/files
4. https://github.com/paritytech/cumulus/pull/663
5. https://github.com/paritytech/cumulus/pull/666
6. https://github.com/paritytech/cumulus/pull/670
7. https://github.com/paritytech/cumulus/pull/657
8. https://github.com/paritytech/cumulus/pull/690

statemine diffs between:

ede4d527c4fc5d84c43216b408a873625488574b -> 0.9.11
f5926c2f50df73bf36171b138241437131d76fff-> 0.9.12

## Tests (Local --> Baikal)
- [x] Does key injection script work (only for calamari-testnet)
- [x] Check block production
- [x] Sudo system.setCode upgrade
- [x] Force transfer balances
- [x] Pass sudo to multi-sig account (try action like adding a Council member)
- [x] Upgrade with authorizeUpgrade + enactAuthorized
- [x] Check average block time is ~12 seconds
- [x] Check dropping and reconnecting a collator
- [ ] Native binary replacement
- [x] Check relay chain XCM message upgrade
- [x] Check governance runtime upgrade

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (`manta` or `manta-pc`) with right title (start with [Manta] or [Manta-PC]),
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests.
- [x] Updated relevant documentation in the code.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [x] If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. If this number is updated, then the spec_version must also be updated 
- [x] If needed, notify the committer this is a draft-release and a tag is needed after merging the PR.
- [ ] Verify benchmarks & weights have been updated for any modified runtime logics

### CODE REVIEW

File: runtime/calamari/src/lib.rs [hunk 1]
```diff
@@ -433,7 +433,7 @@ impl cumulus_pallet_aura_ext::Config for Runtime {}
 
 parameter_types! {
 	pub const KsmLocation: MultiLocation = MultiLocation::parent();
-	pub const RelayNetwork: NetworkId = NetworkId::Polkadot;
+	pub const RelayNetwork: NetworkId = NetworkId::Kusama;
```
Comment (ghzlatarev) [inline runtime/calamari/src/lib.rs:452]: This was overlooked previously.

File: runtime/calamari/src/lib.rs [hunk 2]
```diff
@@ -531,8 +531,8 @@ parameter_types! {
 	pub const MaxDownwardMessageWeight: Weight = MAXIMUM_BLOCK_WEIGHT / 10;
 }
 
-/// No local origins on this chain are allowed to dispatch XCM sends/executions.
-pub type LocalOriginToLocation = SignedToAccountId32<Origin, AccountId, RelayNetwork>;
```
Comment (ghzlatarev) [inline runtime/calamari/src/lib.rs:551]: Old configuration meant that only the relay network's sovereign account could become this origin.
Now we've disabled even that, like Statemine/Statemint.

File: runtime/manta-pc/Cargo.toml [hunk 3]
```diff
@@ -5,7 +5,7 @@ homepage = 'https://manta.network'
 license = 'GPL-3.0'
 name = 'manta-pc-runtime'
 repository = 'https://github.com/Manta-Network/Manta/'
-version = '3.0.8'
+version = '3.0.9'
```
Comment (Dengjianping) [inline runtime/manta-pc/Cargo.toml]: 3.0.9 has been occupied by my pr, please use 3.0.10.

File: runtime/calamari/Cargo.toml [hunk 4]
```diff
@@ -149,6 +150,7 @@ std = [
 	'pallet-membership/std',
 	'manta-primitives/std',
 	'parachain-info/std',
+	#'pallet-asset-tx-payment/std',
```
Comment (ghzlatarev) [inline runtime/calamari/Cargo.toml]: todo: delete this.

File: node/src/service.rs [hunk 5]
```diff
@@ -218,7 +218,7 @@ where
 	sc_client_api::StateBackendFor<TFullBackend<Block>, Block>: sp_api::StateBackend<BlakeTwo256>,
 	Executor: sc_executor::NativeExecutionDispatch + 'static,
 	RB: Fn(
-			Arc<TFullClient<Block, RuntimeApi, Executor>>,
+			Arc<TFullClient<Block, RuntimeApi, NativeElseWasmExecutor<Executor>>>,
```
Comment (stechu) [inline node/src/service.rs]: need to revisit here.

File: runtime/calamari/Cargo.toml [hunk 6]
```diff
@@ -16,65 +16,66 @@ serde = { version = '1.0.119', features = ['derive'], optional = true }
...
+cumulus-primitives-utility = { git = 'https://github.com/paritytech/cumulus.git', default-features = false, branch = "polkadot-v0.9.12" }
+cumulus-pallet-dmp-queue = { git = 'https://github.com/paritytech/cumulus.git', default-features = false, branch = "polkadot-v0.9.12" }
+cumulus-pallet-session-benchmarking = { git = 'https://github.com/paritytech/cumulus.git', default-features = false, optional = true, branch = "polkadot-v0.9.12" }
+cumulus-pallet-xcmp-queue = { git = 'https://github.com/paritytech/cumulus.git', default-features = false, branch = "polkadot-v0.9.12" }
+cumulus-pallet-xcm = { git = 'https://github.com/paritytech/cumulus.git', default-features = false, branch = "polkadot-v0.9.12" }
+pallet-collator-selection = { git = 'https://github.com/paritytech/cumulus.git', default-features = false, branch = "polkadot-v0.9.12" }
+parachain-info = { git = 'https://github.com/paritytech/cumulus.git', default-features = false, branch = "polkadot-v0.9.12" }
+#pallet-asset-tx-payment = { git = 'https://github.com/paritytech/cumulus.git', default-features = false, branch = "polkadot-v0.9.12"}
```
Comment (stechu) [inline runtime/calamari/Cargo.toml]: just delete this if not used?

File: runtime/calamari/src/lib.rs [hunk 7]
```diff
@@ -552,7 +552,7 @@ impl pallet_xcm::Config for Runtime {
 	type SendXcmOrigin = EnsureXcmOrigin<Origin, LocalOriginToLocation>;
 	type XcmRouter = XcmRouter;
 	type ExecuteXcmOrigin = EnsureXcmOrigin<Origin, LocalOriginToLocation>;
-	type XcmExecuteFilter = Everything;
+	type XcmExecuteFilter = Nothing;
```
Comment (ghzlatarev) [inline runtime/calamari/src/lib.rs:573]: @stechu @Dengjianping 
The `nothing` implements the `Contains` trait implements, always returning `false`
Then this check will not pass https://github.com/paritytech/polkadot/blob/master/xcm/pallet-xcm/src/lib.rs#L579
That's why the change was made.
If we want to allow certain locations to pass this filter, we can implement `Contains` ourselves.

File: runtime/calamari/src/lib.rs [hunk 8]
```diff
@@ -615,7 +616,6 @@ impl pallet_session::Config for Runtime {
 	type SessionHandler =
 		<opaque::SessionKeys as sp_runtime::traits::OpaqueKeys>::KeyTypeIdProviders;
 	type Keys = opaque::SessionKeys;
-	type DisabledValidatorsThreshold = DisabledValidatorsThreshold;
```
Comment (ghzlatarev) [inline runtime/calamari/src/lib.rs:618]: @stechu @Dengjianping 
You can review https://github.com/paritytech/substrate/pull/9448 regarding this API change.

**Comment by Dengjianping** [comment]
Remember to update changelog.

File: .github/workflows/publish-draft-releases.yml [hunk 9]
```diff
@@ -305,15 +306,16 @@ jobs:
         if: ${{ needs.check-for-runtime-upgrade.outputs.do-versions-match == 'false' }}
         name: mv and chmod manta-base
         run: |
-          mv ${{ github.workspace }}/manta-base $HOME/.local/bin/
+          ls -ahl ${{ github.workspace }}/
+          mv ${{ github.workspace }}/manta $HOME/.local/bin/manta-base
```
Comment (ghzlatarev) [inline .github/workflows/publish-draft-releases.yml:310]: @stechu after the artifact is unzipped the name is `manta` . So we rename it to `manta-base` when we move it and we can continue without other changes.

**Comment by stechu** [comment]
did a manual test for the upgrade. It passed on my local machine.

**Review by stechu — APPROVED** [review]
